### PR TITLE
MN-714: add configuration and interceptors for rate limiting

### PR DESCRIPTION
**Pull request by MikhailPopov**
**- What I did**
- I have added a configuration for rate-limiting into exporter configuration.
- I have added the 'ServerLimiter' structure, it stores limiters per method per client.
- I have added a temporary implementation of the Limiter interface that always allow execution.

In the next task, we can choose and start to use a suitable limiter implementation instead of the current mock limiter.

Linked PR: https://github.com/insolar/insolar-scripts/pull/14

### CODE REVIEW

File: server/internal/heavy/limits.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func (l *Limiters) GlobalLimit() bool {
+	if l.globalLimiter == nil {
+		return false
+	}
+	return !l.globalLimiter.Allow()
+}
+
+func (l *Limiters) PerClientLimit(ctx context.Context, method string) bool {
```
Comment (Tsovak) [inline server/internal/heavy/limits.go:98]: can we rename the function name to make more readable?  the function returns bool, so, it's just an example, we can rename IsClientDeclined() or HasCleintLimit() and etc

File: server/internal/heavy/limits.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+func (l *ServerLimiter) LimitStream(stream grpc.ServerStream, method string) grpc.ServerStream {
+	return &limitedServerStream{
+		ServerStream: stream,
+		outbound:     l.outbound,
+		method:       method,
+	}
+}
+
+func (l *Limiters) GlobalLimit() bool {
```
Comment (Tsovak) [inline server/internal/heavy/limits.go:91]: and it's also will be nice to rename something as `IsGlobalAllowd()`

**Review by Tsovak — COMMENTED** [review]
comment

File: server/internal/heavy/limits.go (hunk 1 above)
Comment (MikhailPopov) [reply server/internal/heavy/limits.go:98]: Thanks, I will find more readable name.

**Review by Tsovak — APPROVED** [review]
good job there

**Review by LinnikD — APPROVED** [review]

**Review by Tsovak — APPROVED** [review]

File: server/internal/heavy/components_validate_version_test.go [hunk 3]
```diff
@@ -143,7 +143,7 @@ func TestValidateVersionHeavyVersion(t *testing.T) {
 		// test
 		err = validateClientVersion(ctx)
 		require.Error(t, err)
-		require.Contains(t, err.Error(), "version of the observer is outdated. Please upgrade this client")
+		require.Contains(t, err.Error(), "version of the observer is outdated, please upgrade this client")
```
[the remark this answers is not in the document]
Comment (MikhailPopov) [reply server/internal/heavy/components_validate_version_test.go]: Partially done)

**Review by Tsovak — APPROVED** [review]

**Review by uhzxl — APPROVED** [review]

**Review by LinnikD — APPROVED** [review]

**Review by pavlushev — DISMISSED** [review]

**Review by pavlushev — APPROVED** [review]

**Review by uhzxl — APPROVED** [review]

**Review by Tsovak — APPROVED** [review]

**Pull request merged**